frappe/core/doctype/comment/comment.py

```diff
@@ -152,7 +152,10 @@ def get_comments_from_parent(doc):
 	`_comments`
 	"""
 	try:
-		_comments = frappe.db.get_value(doc.reference_doctype, doc.reference_name, "_comments") or "[]"
+		if frappe.db.get_value("DocType", doc.reference_doctype, "is_virtual"):
```

Comment:
Let's stick to this single function, which also implements caching


https://github.com/frappe/frappe/blob/9465314dddbe8d2c17a220fbb1b2c53997a1fc7d/frappe/desk/reportview.py#L737-L739